AWSTranscribeStreaming/Internal/AWSTranscribeStreamingWebSocketProvider.h

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import "AWSTranscribeStreamingWebSocketProviderDelegate.h"
+#import "AWSSRWebSocketDelegateAdaptor.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol AWSTranscribeStreamingWebSocketProvider <NSObject>
+
+@required
+-(void)send:(id)message;
+-(void)connect;
+-(void)disconnect;
+@optional
```

Comment:
If what you mean by optional method? provider can optionally implement?

If the caller does not set the delegate then it's basically saying that they do not care about whether some data is received on the socket (whether it was connected, disconnected, when some data is received on the socket). i believe this is correct behavior.

if the provider can optionally implement the "setDelegate" method, then I don't think it should be optional. the websocket provider should always implement the method so callers can call it. what happens if a websocket provider does not implement setDelegate method and caller calls setDelegate?

Reply:
ya it's optional, its only for Socket Rocket so if your provider doesn't need this you wont need this method so I made it optional so you don't have to have a method just hanging out on your class that you don't need. for native im pretty sure I won't need it.

Reply:
what does the native websocket provider look like? i would understand if setting the delegate is optional, but implementing the setDelegate from the provider's perspective should be required. for example, in a simple case; how does the caller which is holding onto an instance of the native websocket provider which conforms to the new protocol get data from the socket if the provider optionally implemented setDelegate? or to flip it around, if the provider does not implement the setDelegate method, what happens when the caller tries to call setDelegate on the provider?

Reply:
ya I changed it to be required after some more thought